sky/clouds/service_catalog/kubernetes_catalog.py

```diff
@@ -26,3 +29,66 @@ def get_image_id_from_tag(tag: str, region: Optional[str]) -> Optional[str]:
 def is_image_tag_valid(tag: str, region: Optional[str]) -> bool:
     """Returns whether the image tag is valid."""
     return common.is_image_tag_valid_impl(_image_df, tag, region)
+
+
+def list_accelerators(
+        gpus_only: bool,
+        name_filter: Optional[str],
+        region_filter: Optional[str],
+        quantity_filter: Optional[int],
+        case_sensitive: bool = True
+) -> Dict[str, List[common.InstanceTypeInfo]]:
+    has_gpu = kubernetes_utils.detect_gpu_resource()
```

Comment:
If the cluster is down or Kubernetes is not enabled in SkyPilot, `sky show-gpus` fails with:
```
Traceback (most recent call last):
  File "/Users/romilb/tools/anaconda3/bin/sky", line 8, in <module>
    sys.exit(cli())
  File "/Users/romilb/tools/anaconda3/lib/python3.9/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/Users/romilb/tools/anaconda3/lib/python3.9/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/utils/common_utils.py", line 311, in _record
    return f(*args, **kwargs)
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/cli.py", line 1162, in invoke
    return super().invoke(ctx)
  File "/Users/romilb/tools/anaconda3/lib/python3.9/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/romilb/tools/anaconda3/lib/python3.9/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/romilb/tools/anaconda3/lib/python3.9/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/clouds/service_catalog/config.py", line 51, in wrapper
    return func(*args, **kwargs)
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/utils/common_utils.py", line 332, in _record
    return f(*args, **kwargs)
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/cli.py", line 3388, in show_gpus
    for out in _output():
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/cli.py", line 3309, in _output
    result = service_catalog.list_accelerators(gpus_only=True,
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/clouds/service_catalog/config.py", line 51, in wrapper
    return func(*args, **kwargs)
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/clouds/service_catalog/__init__.py", line 66, in list_accelerators
    results = _map_clouds_catalog(clouds, 'list_accelerators', gpus_only,
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/clouds/service_catalog/__init__.py", line 43, in _map_clouds_catalog
    results.append(method(*args, **kwargs))
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/clouds/service_catalog/kubernetes_catalog.py", line 41, in list_accelerators
    has_gpu = kubernetes_utils.detect_gpu_resource()
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/utils/kubernetes_utils.py", line 192, in detect_gpu_resource
    nodes = get_kubernetes_nodes()
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/utils/kubernetes_utils.py", line 205, in get_kubernetes_nodes
    nodes = kubernetes.core_api().list_node(
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/adaptors/kubernetes.py", line 42, in wrapper
    return func(*args, **kwargs)
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/adaptors/kubernetes.py", line 93, in core_api
    _load_config()
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/adaptors/kubernetes.py", line 42, in wrapper
    return func(*args, **kwargs)
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/adaptors/kubernetes.py", line 85, in _load_config
    raise ValueError(err_str) from None
ValueError: Failed to load Kubernetes configuration. Please check if your kubeconfig file is valid. Error: Invalid kube-config file. No configuration found.
```

We may need to add a check if Kubernetes cluster is enabled and possibly call `kubernetes_utils.check_credentials` to verify the health of the cluster.

Reply:
Addressed in https://github.com/skypilot-org/skypilot/pull/2638/commits/6b1691e445c3c711a3d93fb5d21da3912d452359